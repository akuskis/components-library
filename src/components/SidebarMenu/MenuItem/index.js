// @flow

import React from 'react';
import { NavLink } from 'react-router-dom';

import './menuItem.css';

type Props = {
  title: string,
  routePath: string
};

const MenuItem = ({ title, routePath }: Props) => (
  <NavLink to={routePath} className="menu-item" activeClassName="menu-item_active">
    <div>{title}</div>
    <div className="menu-item__trailing-line" />
  </NavLink>
);

export default MenuItem;
