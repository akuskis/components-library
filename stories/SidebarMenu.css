.sidebar-menu-story {
  display: flex;
  flex-direction: row;
}

.sidebar-menu-story__content {
  width: 300px;
}

.sidebar-menu-story__vertical-line {
  border-left: 1px solid #adadaf;
}
