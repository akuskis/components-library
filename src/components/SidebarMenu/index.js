// @flow

import React from 'react';

import MenuItem from './MenuItem';
import './sidebarMenu.css';

type Props = {
  items: Array<{
    title: string,
    routePath: string
  }>
};

const SidebarMenu = ({ items }: Props) => {
  const content =
    items && items.map(item => <MenuItem key={item.title + item.routePath} {...item} />);

  return <div className="sidebar-menu">{content}</div>;
};

export default SidebarMenu;
