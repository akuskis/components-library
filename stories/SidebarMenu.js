import React from 'react';
import StoryRouter from 'storybook-react-router';
import { storiesOf } from '@storybook/react';

import SidebarMenu from '../src/components/SidebarMenu';

storiesOf('SidebarMenu', module)
  .addDecorator(StoryRouter())
  .add('basic', () => {
    const menuItems = [
      { title: 'Item A', routePath: '/a' },
      { title: 'Item B', routePath: '/b' },
      { title: 'Item C', routePath: '/c' }
    ];

    return <SidebarMenu items={menuItems} />;
  });
