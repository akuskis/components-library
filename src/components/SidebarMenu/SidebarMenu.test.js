import React from 'react';
import { shallow } from 'enzyme';

import SidebarMenu from '.';

describe('<SidebarMenu />', () => {
  it('should render without props', () => {
    const component = shallow(<SidebarMenu />);

    expect(component).toMatchSnapshot();
  });

  it('should render with content', () => {
    const items = [
      { title: 'Item A', routePath: '/a' },
      { title: 'Item B', routePath: '/b' },
      { title: 'Item C', routePath: '/c' }
    ];

    const component = shallow(<SidebarMenu items={items} />);

    expect(component).toMatchSnapshot();
  });
});
