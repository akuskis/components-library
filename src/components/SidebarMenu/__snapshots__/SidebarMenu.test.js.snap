// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SidebarMenu /> should render with content 1`] = `
<div
  className="sidebar-menu"
>
  <MenuItem
    key="Item A/a"
    routePath="/a"
    title="Item A"
  />
  <MenuItem
    key="Item B/b"
    routePath="/b"
    title="Item B"
  />
  <MenuItem
    key="Item C/c"
    routePath="/c"
    title="Item C"
  />
</div>
`;

exports[`<SidebarMenu /> should render without props 1`] = `
<div
  className="sidebar-menu"
/>
`;
